lib/bap_disasm/bap_disasm_x86_lifter.ml

```diff
@@ -0,0 +1,2319 @@
+(** Native lifter of x86 instructions to the BAP IL *)
+
+open Core_kernel.Std
+open Bap_types.Std
+(* open Bil *)
+module BZ = Big_int_Z
+(* open Type *)
+(* open Arch *)
+(* open Var *)
+
+module BV = Bitvector
+
+module Util = struct
+  let id x = x (* useful for tabulating lists, for instance. *)
+
+  (* some of the "imported" zarith-processing functions use this. *)
+  (* XXX we don't like mutable state. Bitvector should be written so that the 2 big_int_Z
+   * functions that use this can be killed. *)
+
+  let index_ofq elt lst =
+    match List.findi lst ~f:(fun _ e -> e = elt) with
+    | None -> raise Not_found
+    | Some (i,_) -> i
+
+  let concat_explist elist =
+    List.reduce_exn
+      ~f:Exp.(^) elist
+end
+
+module Strip = struct
+  let bits_of_width typ =
+    let open Type in
+    match typ with
+    | Imm n -> n
+    | Mem _ -> invalid_arg "bits_of_width"
+  let bytes_of_width t =
+    let b = bits_of_width t in
+    if not ((b mod 8) = 0) then invalid_arg "bytes_of_width";
+    b / 8
+end
+open Strip
+
+let exp_false = Exp.int BV.b0
+let exp_true  = Exp.int BV.b1
+let exp_not = Exp.lnot
+
+(* XXX this is a lot of crap copied from arithmetic.ml
+ * I'm starting to think we should just pull in the old
+ * big_int code instead. unless I'm missing something and this stuff
+ * is slated to go away? *)
+let it n t = BV.of_int n ~width:t |> Exp.int
+
+module Big_int_temp = struct
```

Comment:
this should be removed. You should use `Bitvector` module. Look at the `Integer` interface.